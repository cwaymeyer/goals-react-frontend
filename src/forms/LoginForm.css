.LoginForm {
  background-color: rgb(244, 248, 252);
}
