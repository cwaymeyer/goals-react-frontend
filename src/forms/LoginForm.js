import { useState } from "react";

function LoginForm({ login }) {
  const initialState = {
    username: "",
    password: "",
  };

  const [formData, setFormData] = useState(initialState);

  function handleChange(e) {
    const { name, value } = e.target;
    setFormData((formData) => ({ ...formData, [name]: value }));
  }

  async function handleSubmit(e) {
    console.log(formData);
    e.preventDefault();
    await login(formData);
  }

  return (
    <div className="LoginForm">
      <h1>This is the login form</h1>
      <form onSubmit={handleSubmit}>
        <input
          type="text"
          name="username"
          placeholder="username"
          value={formData.username}
          onChange={handleChange}
          required
        />
        <input
          type="password"
          name="password"
          placeholder="password"
          value={formData.password}
          onChange={handleChange}
          required
        />
        <button type="submit" onSubmit={handleSubmit}>
          Log in
        </button>
      </form>
    </div>
  );
}

export default LoginForm;
