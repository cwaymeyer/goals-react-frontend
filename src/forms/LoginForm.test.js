import { MemoryRouter } from "react-router";
import { render, screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import LoginForm from "./LoginForm";

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <LoginForm />
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <LoginForm />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("displays correctly", function () {
  const { getByTestId, getByText } = render(
    <MemoryRouter>
      <LoginForm />
    </MemoryRouter>
  );
  expect(getByText("Log in to account")).toBeInTheDocument();
  expect(getByTestId("login-form")).toBeInTheDocument();
});
