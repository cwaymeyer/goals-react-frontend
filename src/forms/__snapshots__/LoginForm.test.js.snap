// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="LoginForm"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-colorDisabled MuiSvgIcon-fontSizeMedium css-nf1lt7-MuiSvgIcon-root"
        data-testid="FitnessCenterIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20.57 14.86 22 13.43 20.57 12 17 15.57 8.43 7 12 3.43 10.57 2 9.14 3.43 7.71 2 5.57 4.14 4.14 2.71 2.71 4.14l1.43 1.43L2 7.71l1.43 1.43L2 10.57 3.43 12 7 8.43 15.57 17 12 20.57 13.43 22l1.43-1.43L16.29 22l2.14-2.14 1.43 1.43 1.43-1.43-1.43-1.43L22 16.29z"
        />
      </svg>
      <div
        class="MuiBox-root css-nub3x0"
      >
        <div
          class="MuiBox-root css-14fqnlk"
        >
          <h1>
            GOALS
          </h1>
        </div>
        <div
          class="MuiBox-root css-bk9fzy"
        >
          <p>
            Track your workout progress
          </p>
        </div>
      </div>
    </div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthXs css-nzinn5-MuiContainer-root"
    >
      <div
        class="MuiBox-root css-n3fmse"
      >
        <h1>
          Log in to account
        </h1>
      </div>
      <form
        data-testid="login-form"
      >
        <div
          class="MuiGrid-root MuiGrid-container css-1cs2yco-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1qobofj-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiTextField-root css-136h9lv-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="mui-3"
                id="mui-3-label"
              >
                Username
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="mui-3"
                  name="username"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-173wfxe"
                  >
                    <span>
                      Username
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1qobofj-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiTextField-root css-136h9lv-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="mui-4"
                id="mui-4-label"
              >
                Password
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="mui-4"
                  name="password"
                  type="password"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-173wfxe"
                  >
                    <span>
                      Password
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root css-0"
          />
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root  css-195sjap-MuiButtonBase-root-MuiButton-root"
            data-testid="login-form-submit"
            tabindex="0"
            type="submit"
          >
            Log in
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-veflwg-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Sign up for new account
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </form>
    </div>
    <div
      class="info"
    >
      This app created by 
      <b>
        Caleb Waymeyer
      </b>
      . Check out the source code 
      <a
        href="https://github.com/cwaymeyer/goals-react-frontend"
        target="_blank"
      >
        here
      </a>
      .
    </div>
  </div>
</DocumentFragment>
`;
